### PR TITLE
Add output stream to MessageBuilder

### PR DESCRIPTION
**Pull request by TsundereBug**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * Message in #testing

**Issues Fixed:** Previously couldn't use an output stream to setup a file to send.

### Changes Proposed in this Pull Request
* Add a `fileOutputStream(String fileName)` method that gives you an output stream you can write to.

### CODE REVIEW

**Comment by GrandPanda** [comment]
MessageBuilder needs to be modified to allow for multiple file uploads at some point. I don't see how this can work when that happens.

**Comment by TsundereBug** [comment]
Make it a Set of output streams and map those to a list of input streams and add the sets. Each time calling the method gives a new output stream.

**Comment by MinnDevelopment** [comment]
Taking the fact aside how mediocre this implementation is, what advantage would this feature give over the input stream variant? Can you give an example?

**Comment by TsundereBug** [comment]
Appending data on the go to upload as a file. Dumping large data is a great example, or if doing program evaluations.

File: src/main/java/sx/blah/discord/util/MessageBuilder.java [hunk 1]
```diff
@@ -233,10 +232,24 @@ public MessageBuilder withFile(File file) throws FileNotFoundException {
...
 
 	/**
+	 * Gives an output stream you can write file data to.
+	 *
+	 * @param fileName the name of the file to be sent to Discord
+	 * @return an OutputStream to write file data to.
+	 */
+	public OutputStream fileOutputStream(String fileName) {
+		this.fileName = fileName;
+		this.outputStream = new ByteArrayOutputStream();
```
Comment (ArsenArsen) [inline src/main/java/sx/blah/discord/util/MessageBuilder.java]: Close the previous one if it exists

File: src/main/java/sx/blah/discord/util/MessageBuilder.java [hunk 2]
```diff
@@ -233,10 +232,24 @@ public MessageBuilder withFile(File file) throws FileNotFoundException {
 	public MessageBuilder withFile(InputStream stream, String fileName) {
 		this.stream = stream;
 		this.fileName = fileName;
+		this.useOutput = false;
```
Comment (ArsenArsen) [inline src/main/java/sx/blah/discord/util/MessageBuilder.java]: Can't you just check `!= null`?

**Comment by austinv11** [comment]
It's fundamentally wrong to use an OutputStream due to the intended directionality of the data since MessageBuilder needs to read data an InputStream would be best suited. Plus, as you've demonstrated, it's a trivial task to even send an OutputStream in the case it is in fact necessary.

**Pull request closed without merging**